### ISSUE TITLE
Add a ban command

### ISSUE THREAD

**Issue by tornac1234** [SubnauticaNitrox/Nitrox#1734]
We need a ban command for the future.

**Comment by Measurity** [comment]
I would start with IP bans. If it's LAN, just ignore banning because it should be easy for server owner to manage with password.

Maybe in future we can add options for more secure authentication. Like requiring a google account with payment info (without actually sending payment info to server).

**Comment by RokeJulianLockhart** [comment]
You're right, LAN bans are totally unenforceable – NAT makes consistent IP identification impossible, and MAC address randomization (only possible on LANs) doesn't work anymore.

**Comment by SmokerDoker** [comment]
Some thoughts on a concrete design for this, based on how the existing command/permission/networking code is structured:

**Enforcement point:** IP bans should be checked in `PlayerManager.ReservePlayerContext` (called from `MultiplayerSessionReservationRequestProcessor`), which already validates password/capacity/name conflicts and returns a `MultiplayerSessionReservationState` with a user-facing `[Description]` (e.g. the existing `AUTHENTICATION_FAILED` message). A new `PLAYER_BANNED` flag would reuse that same mechanism to show the banned player a proper reason, instead of just dropping the raw connection with no explanation.

**Storage:** A dedicated persisted ban list (its own file, e.g. `bans.json`, alongside the existing per-server save data), keyed primarily by IP address, with player name/PeerId kept alongside for readability and `unban <name>` convenience. Since IP↔player mapping today only lives in-memory for the current server run (`SessionManager`), a player's last-known IP would also need to be persisted (e.g. added to `PersistedPlayerData`, which already persists other per-player moderation state like `Permissions`) so offline players can still be banned by name, not just by raw IP.

**Both temporary and permanent bans from the start:** each ban entry would carry an optional expiry timestamp — `null`/absent for a permanent ban, a concrete time for a temp ban. The reservation check simply treats an expired entry as not-banned (and can lazily prune it), so we don't need separate command paths for temp vs. permanent — just an optional duration argument on the same `ban` command.

**Commands** (following the existing `KickCommand` pattern — `[RequiresPermission(Perms.MODERATOR)]`, same self-ban/rank-check guards):
- `ban <player|ip> [reason] [duration]` — bans (permanent if no duration given), and immediately kicks the player if currently online.
- `unban <player|ip>`
- `banlist` — lists active bans (mods+ only)

**Known limitation (already discussed above):** this only ever blocks by IP, so it remains bypassable via VPN/dynamic IP/shared NAT, same as acknowledged in the thread already. That's an accepted tradeoff here, not something this design tries to solve — a stronger auth layer would be a separate, much bigger effort.

Happy to turn this into a PR if there's agreement on the approach.

**Comment by Measurity** [comment]
@SmokerDoker sounds good to me, thanks for writing down the plan.

FYI. I've done some cleanup to the reservation code in #2805 that might affect this work. It involves deprecating `reservationId` and `correlationId` as we use `SessionId` only now, simplifying things. 

Up to you if you want to work on the ban command before this PR is merged.

**Comment by SmokerDoker** [comment]
Sounds good, I’ll hold off on the ban command until #2805 is merged to avoid building on top of correlationId/reservationId right as they’re being removed. Will pick it up once that’s in.
